src/Browscap/Data/Factory/DataCollectionFactory.php

```diff
@@ -0,0 +1,272 @@
+        }
+    }
+
+    /**
+     * Load a JSON file, parse it's JSON to arrays, validates these data and creates a
+     * collection of Division objects
+     *
+     * @param string $filename Name of the file
+     *
+     * @throws \RuntimeException         If the file does not exist or has invalid JSON
+     * @throws \UnexpectedValueException If required attibutes are missing in the division
+     * @throws \LogicException
+     */
+    public function addSourceFile(string $filename) : void
+    {
+        $divisionData = $this->loadFile($filename);
+
+        $this->divisionDataValidator->validate($divisionData, $filename, $this->allDivisions, false);
+
+        $this->collection->addDivision(
+            $this->divisionFactory->build(
+                $divisionData,
+                $filename,
+                false
+            )
+        );
+    }
+
+    /**
+     * Load the file for the default properties
+     *
+     * @param string $filename Name of the file
+     *
+     * @throws \RuntimeException if the file does not exist or has invalid JSON
+     */
+    public function setDefaultProperties(string $filename) : void
+    {
+        $divisionData = $this->loadFile($filename);
+        $this->divisionDataValidator->validate($divisionData, $filename, $this->allDivisions, true);
+
+        $this->collection->setDefaultProperties(
+            $this->divisionFactory->build(
+                $divisionData,
+                $filename,
+                true
+            )
+        );
+    }
+
+    /**
+     * Load the file for the default browser
+     *
+     * @param string $filename Name of the file
+     *
+     * @throws \RuntimeException if the file does not exist or has invalid JSON
+     */
+    public function setDefaultBrowser(string $filename) : void
+    {
+        $divisionData = $this->loadFile($filename);
+        $this->divisionDataValidator->validate($divisionData, $filename, $this->allDivisions, true);
+
+        $this->collection->setDefaultBrowser(
+            $this->divisionFactory->build(
+                $divisionData,
+                $filename,
+                true
+            )
+        );
+    }
+
+    /**
+     * @param string $filename
+     *
+     * @throws \RuntimeException
+     *
+     * @return array
+     */
+    private function loadFile(string $filename) : array
+    {
+        if (!file_exists($filename)) {
+            throw new \RuntimeException('File "' . $filename . '" does not exist.');
+        }
+
+        Assertion::readable($filename, 'File "' . $filename . '" is not readable.');
+
+        $fileContent = file_get_contents($filename);
+
+        Assertion::string($fileContent);
+
+        if (preg_match('/[^ -~\s]/', $fileContent)) {
+            throw new \RuntimeException('File "' . $filename . '" contains Non-ASCII-Characters.');
+        }
+
+        $decodedFileContent = json_decode($fileContent, true);
+
+        if (null === $decodedFileContent) {
```

Comment:
For JSON parsing, it might be worth looking into [seldaek/jsonlint](https://github.com/seldaek/jsonlint) which provides more comprehensive JSON linting - I'd add that this is outside the scope of this PR though, we can just create an issue to add this later.

Reply:
We have a lot of dulicate keys in the json files. So we have to do this later.